src/kili/services/label_data_parsing/json_response.py

```diff
@@ -1,34 +1,81 @@
+        self._json_data: Dict[str, "FramesList"] = {}
 
-        This class will modify the input json_response.
-        If you want to keep the original json_response, use deepcopy.
+        self._nb_frames = len(json_response)
 
-        Args:
-            project_info: Information about the project.
-            json_response: Value of the key "jsonResponse" of a label.
-            job_names_to_parse: List of job names to parse. By default, parse all the jobs that are not children.
-        """
+        json_interface = project_info["jsonInterface"]
+
+        # all job names in the json response should be in the json interface too
+        for _, frame_response in json_response.items():
+            for job_name in frame_response:
+                if job_name not in json_interface:
+                    raise JobNotExistingError(job_name)
+
+        # define the list of job names to parse
+        # by default, parse all the jobs that are not children
+        if job_names_to_parse is None:
+            job_names_to_parse = [
+                job_name
+                for job_name, job_interface in json_interface.items()
+                if not job_interface["isChild"]
+            ]
+
+        for current_job_name in job_names_to_parse:
+            frames_list_for_job = []
+            for _, frame_json_response in sorted(
+                json_response.items(), key=lambda item: int(item[0])  # sort by frame number
+            ):
+                job_response = frame_json_response.get(current_job_name, {})
+                job_payload = job_response_module.JobPayload(
+                    job_name=current_job_name,
+                    project_info=project_info,
+                    job_payload=job_response,
+                )
+                frames_list_for_job.append(job_payload)
+
+            assert len(frames_list_for_job) == self._nb_frames, (
+                f"len(frames_list_for_job) = {len(frames_list_for_job)} != {self._nb_frames} ="
+                " self._nb_frames"
+            )
+            self._json_data[current_job_name] = FramesList(frames_list_for_job)
+
+    def to_dict(self) -> Dict[str, Dict]:
+        """Returns a copy of the parsed label as a dict."""
+        ret = {str(frame_id): {} for frame_id in range(self._nb_frames)}
+        for job_name, frames_list in self._json_data.items():
+            for frame_id, job_payload in enumerate(frames_list):
+                job_payload_dict = job_payload.to_dict()
+                if job_payload_dict:
+                    ret[str(frame_id)][job_name] = job_payload_dict
+        return ret
+
+
+# pylint: disable=too-few-public-methods
+class _ParsedOtherJobs:
+    def __init__(
+        self,
+        project_info: Project,
+        json_response: Dict,
+        job_names_to_parse: Optional[List[str]] = None,
+    ) -> None:
         self._json_data: Dict[str, "job_response_module.JobPayload"] = {}
 
         json_interface = project_info["jsonInterface"]
```

Comment:
comment for a previous PR
in ParseOtherJobs, you test:
```
# a required parent job should have a non-empty json response
            if (
                not job_interface["isChild"]  # check if parent
                and job_interface["required"]  # check if required
                and "VIDEO" not in project_info["inputType"]  # can have empty frames
                and not job_response
            ):
                raise JobNotExistingError(job_name)
```
But we agreed with PM when I was working on json response validation that one can import a label from the API/SDK with empty response for required jobs. It is very useful for predictions labels for example when you want to upload predictions only for some jobs. But in the general case, the required feature only work when labeling on the platform.
It should be changed. Can be in another PR

Reply:
oh ok I didn't know that, thanks